### PR TITLE
Logs improvements

### LINKED ISSUE #14: Logs pagination doesn't save per page value
When selecting a different value for the amount of logs per page the change doesn't get reflected by the dropdown and when changing the page it gets reset back to 100
grafik

### PR DESCRIPTION
**Pull request by Geekyhobo**
### New Logs Hurray
Logs have been improved, with the ability to search for all values has been added the look and feel have also been improved.
css is hell and hurt my brain.

## Some Things to Note
- The side panel with more details now can be opened with only one click
- Side panel now links to player pages if the log message includes a player
- Logtype has an autocomplete so you don't have to remember the 90 Logtypes
- Severity dropdown has fAnCy colors
- Server cannot be used currently used :( I will fix it later but its beyond the scope of this PR, TLDR server is not stored with the admin-log table in the database
- The calendar selector has been made a bit more intuitive
- No more drop down filter hell

Huge thanks to Julian for helping me with my razor pages woes and java script stuff

Fixes space-wizards/space-station-14#10677
Fixes #14 

## Pictures
New logs
Screenshot 2023-12-22 013219

Side Panel
Screenshot 2023-12-22 013407

Autocomplete
Screenshot 2023-12-22 013301

Severity drop down
Screenshot 2023-12-22 013159

### CODE REVIEW

File: .idea/.idea.SS14.Admin/.idea/.name [hunk 1]
```diff
@@ -1 +0,0 @@
-SS14.Admin
```
Comment (Geekyhobo) [inline .idea/.idea.SS14.Admin/.idea/.name:1]: No Clue why or how this was deleted someone tell me if its important and ill add it back

Comment (DrSmugleaf) [reply .idea/.idea.SS14.Admin/.idea/.name:1]: The solution to this is to add `.idea/` to the .gitignore

**Review by DrSmugleaf — APPROVED** [review]
With the .gitignore change this should be good to merge

File: .idea/.idea.SS14.Admin/.idea/.name (hunk 1 above)
Comment (Geekyhobo) [reply .idea/.idea.SS14.Admin/.idea/.name:1]: Fixed, added .name back and to my .gitignore cheers

File: SS14.Admin/wwwroot/js/logs-grid.js [hunk 2]
```diff
@@ -2,23 +2,18 @@
 const overlay = document.getElementById('side-panel-overlay');
 const entries = document.querySelectorAll('.data-log');
 
-let selecting = false;
-
-document.addEventListener('selectstart', () => selecting = true);
```
Comment (juliangiebel) [inline SS14.Admin/wwwroot/js/logs-grid.js:7]: The selecting stuff was for allowing you to select text inside the table.
I guess that is what caused opening the details panel to be finicky.

**Review by juliangiebel — APPROVED** [review]
Thank you for working on the admin logs page and implementing the mockup. 👍 
This looks good to merge.

File: SS14.Admin/Pages/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -13,6 +13,8 @@
     <title>@ViewData["Title"] - SS14.Admin</title>
     <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css"/>
     <link rel="stylesheet" href="~/css/site.css"/>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/all.min.css">
```
Comment (PJB3005) [inline SS14.Admin/Pages/Shared/_Layout.cshtml:16]: We definitely should not rely on a CDN for stuff like this, we should bundle it with the program itself.

Comment (Geekyhobo) [reply SS14.Admin/Pages/Shared/_Layout.cshtml:16]: I’ll fix it when I get home

File: SS14.Admin/wwwroot/js/log-type-autocomplete.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
+﻿document.addEventListener('DOMContentLoaded', function () {
+    const LogType = {
```
Comment (PJB3005) [inline SS14.Admin/wwwroot/js/log-type-autocomplete.js:2]: This list is quickly going to get out of date. It'd be best if SS14.Admin generates this list from the raw enum definition.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
I'll fix the issues myself real quick.

**Pull request merged**